File: frontend/src/data/mock/tax.ts
export interface AccountComparison {
  id: string;
  name: string;
  tag: string;
  tax: string;
  limit: string;
  product: string;
  note: string;
  score: number;
}

export const accounts: AccountComparison[] = [
  { id: 'direct', name: '해외주식 직투', tag: '유동성', tax: '연 250만원 공제 후 22%', limit: '제한 없음', product: 'QQQM', note: '달러 자산을 직접 보유하고 언제든 매매하기 좋아요.', score: 76 },
  { id: 'isa', name: '중개형 ISA', tag: '절세', tax: '200만원 비과세 + 9.9%', limit: '연 2,000만원', product: '국내상장 해외 ETF', note: '3년 이상 투자하고 목돈을 운용할 때 효율적이에요.', score: 88 },
  { id: 'pension', name: '연금저축펀드', tag: '장기투자', tax: '최대 16.5% 세액공제', limit: '세액공제 연 600만원', product: 'TIGER 미국나스닥100', note: '55세 이후 사용할 장기 자금이라면 우선순위가 높아요.', score: 96 },
  { id: 'irp', name: 'IRP', tag: '노후', tax: '연금저축 합산 900만원', limit: '연 1,800만원', product: 'ETF + 안전자산 30%', note: '추가 세액공제에 좋지만 중도인출 제약을 확인하세요.', score: 84 },
];

File: backend/app/schemas/tax.py
from decimal import Decimal
from typing import Literal

from pydantic import BaseModel, Field, model_validator

AccountType = Literal["direct", "isa", "pension", "irp"]


class TaxSimulationRequest(BaseModel):
    annual_salary_krw: Decimal = Field(ge=0, le=Decimal("10000000000"))
    monthly_contribution_krw: Decimal = Field(gt=0, le=Decimal("100000000"))
    investment_years: int = Field(ge=3, le=40)
    annual_return_rate_pct: Decimal = Field(ge=Decimal("-100"), le=Decimal("30"))
    withdrawal_age: int = Field(default=60, ge=55, le=100)


class TaxRuleSource(BaseModel):
    title: str
    url: str
    authority: str


class TaxRuleSummary(BaseModel):
    version: str
    effective_date: str
    parameters: dict[str, str]
    sources: list[TaxRuleSource]


class TaxAccountResult(BaseModel):
    account_type: AccountType
    name: str
    tag: str
    tax_description: str
    contribution_limit_description: str
    recommended_product: str
    eligible_contribution: Decimal
    overflow_contribution: Decimal
    gross_value: Decimal
    contribution_tax_credit: Decimal
    investment_tax: Decimal
    withdrawal_tax: Decimal
    after_tax_value: Decimal
    tax_savings_vs_direct: Decimal
    score: int
    benefits: list[str]
    cautions: list[str]


class TaxSimulationResponse(BaseModel):
    best_account_type: AccountType
    results: list[TaxAccountResult]
    rules: TaxRuleSummary
    assumptions: list[str]
    disclaimer: str


class PensionStartComparisonRequest(BaseModel):
    annual_salary_krw: Decimal = Field(ge=0, le=Decimal("10000000000"))
    current_age: int = Field(ge=18, le=79)
    withdrawal_age: int = Field(ge=55, le=100)
    monthly_contribution_krw: Decimal = Field(gt=0, le=Decimal("100000000"))
    annual_return_rate_pct: Decimal = Field(ge=Decimal("-100"), le=Decimal("30"))
    delay_years: int = Field(default=5, ge=1, le=20)

    @model_validator(mode="after")
    def validate_age_window(self) -> "PensionStartComparisonRequest":
        if self.current_age >= self.withdrawal_age:
            raise ValueError("현재 나이는 연금 수령 나이보다 작아야 합니다.")
        if self.current_age + self.delay_years >= self.withdrawal_age:
            raise ValueError("늦춘 시작 나이는 연금 수령 나이보다 작아야 합니다.")
        return self


class PensionStartScenario(BaseModel):
    start_age: int
    contribution_years: int
    annual_eligible_contribution: Decimal
    total_principal: Decimal
    projected_balance: Decimal
    contribution_tax_credit: Decimal
    projected_value_with_tax_credit: Decimal


class PensionStartComparisonResponse(BaseModel):
    start_now: PensionStartScenario
    delayed_start: PensionStartScenario
    projected_value_gap: Decimal
    delayed_required_monthly_contribution: Decimal
    delayed_required_within_pension_limit: bool
    rules: TaxRuleSummary
    assumptions: list[str]
    disclaimer: str
